returnn/datasets/meta.py

```diff
@@ -317,14 +320,13 @@ def _load_seq_list(self, seq_list_file=None):
     return seq_list
 
   def _get_dataset_seq_length(self, seq_idx):
+    default_dataset = self.datasets[self.default_dataset_key]
     if not self.orig_seq_order_is_initialized:
       # To use get_seq_length() we first have to init the sequence order once in original order.
       # If sequence lengths are not needed by get_seq_order_for_epoch this is never executed.
-      self.datasets[self.default_dataset_key].init_seq_order(
-        epoch=self.epoch, seq_list=self.seq_list_original[self.default_dataset_key])
+      default_dataset.init_seq_order(epoch=self.epoch, seq_list=self.seq_list_original[self.default_dataset_key])
       self.orig_seq_order_is_initialized = True
-
-    return self.datasets[self.default_dataset_key].get_seq_length(seq_idx)["data"]
+    return default_dataset.get_seq_length(seq_idx)[default_dataset.default_data_key]
```

Comment:
If you use that here, I think all the usages of `get_seq_order_for_epoch` should be adopted as well to use it, to keep this consistent.

Reply:
This is exactly the reason why I made "default_data_key" configurable for the MetaDataset only, the possible amount of usages (which will not only be for the sequence order) is huge.

If you say this should be changed consistently all at once, I can do that, but this may take several days.

Reply:
`default_dataset` here is probably never a `MetaDataset`. I don't see how the whole approach is useful then.